r/R/dataset.R

```diff
@@ -321,12 +325,10 @@ FileSystemSourceFactory$create <- function(filesystem,
 #' @export
 FileFormat <- R6Class("FileFormat", inherit = Object)
 FileFormat$create <- function(format, ...) {
```

Comment:
I would do this a little differently: 

```suggestion
FileFormat$create <- function(format, ..., options = list(...)) {
```

and pass `options` in L329 below. Then you can remove the `reader_options` hack in dataset.cpp. This also should be documented.